### PR TITLE
add Clipper

### PR DESCRIPTION
**Pull request by SimonDanisch**

### CODE REVIEW

**Comment by giordano** [comment]
This is missing the license file:
```
┌ Warning: Unable to find valid license file in "${prefix}/share/licenses/Clipper"
└ @ BinaryBuilder ~/.julia/dev/BinaryBuilder/src/Auditor.jl:438
```

File: C/Clipper/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+${CXX} -fPIC -std=c++11 -shared -o "${libdir}/libcclipper.${dlext}" clipper.cpp cclipper.cpp
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libcclipper", :cclipper)
```
Comment (giordano) [inline C/Clipper/build_tarballs.jl:27]: I renamed the product to use the more standard `lib*` prefix

**Comment by jaakkor2** [comment]
Not a blocker, but Clipper version should likely not be v1.0.0.  See https://github.com/SimonDanisch/ClipperBuilder/issues/2.
As I do not use this library, I do not care too much whatever is decided.

File: C/Clipper/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+name = "Clipper"
+version = v"1.0.0"
```
Comment (staticfloat) [inline C/Clipper/build_tarballs.jl:6]: Looking at the source it looks like it should be `v6.4.0`.

File: C/Clipper/build_tarballs.jl (hunk 1 above)
Comment (staticfloat) [reply C/Clipper/build_tarballs.jl:27]: I don't see that; but I think the variable most likely should be `libcclipper` as well, although if that's not how Simon wants to organize it, that's fine.  Just a suggestion for uniformity across the ecosystem.

File: C/Clipper/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    GitSource("https://github.com/SimonDanisch/ClipperBuilder.git",
+              "9ea1878235d518d5e14bce35a94924a34bab8b68"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/ClipperBuilder/
+mkdir "${libdir}"
+${CXX} -fPIC -std=c++11 -shared -o "${libdir}/libcclipper.${dlext}" clipper.cpp cclipper.cpp
+"""
```
Comment (staticfloat) [inline C/Clipper/build_tarballs.jl:19]: The license that should be installed is http://www.boost.org/LICENSE_1_0.txt

I suggest downloading and putting that into the `ClipperBuilder` repository, then you can just `install_license LICENSE_1_0.txt`.

File: C/Clipper/build_tarballs.jl (hunk 1 above)
Comment (SimonDanisch) [reply C/Clipper/build_tarballs.jl:27]: I dont really care :D Consistency across the ecosystem seems desirable, though!

File: C/Clipper/build_tarballs.jl (hunk 3 above)
Comment (SimonDanisch) [reply C/Clipper/build_tarballs.jl:19]: Ok, I updated the sha to include:
https://github.com/SimonDanisch/ClipperBuilder/commit/d9e011161e5f4e161137af19d0f5da6dc5764520

**Pull request merged**